src/CacheAdapters/RedisCacheItemPoolAdapter.php

```diff
@@ -126,7 +126,13 @@ public function deleteItems(array $keys): bool
      */
     public function save(CacheItemInterface $item)
     {
-        return $this->set($item->getKey(), serialize($item->get()), ($this->getItemLifeTime)($item));
+        $ttl = ($this->getItemLifeTime)($item);
+        if ($ttl === null)
+        {
+            return $this->set($item->getKey(), serialize($item->get()));
+        } else {
+            return $this->setex($item->getKey(), $ttl, serialize($item->get()));
+        }
```

Comment:
The else statement is avoidable in this case ;)

```suggestion
        $ttl = ($this->getItemLifeTime)($item);
        if ($ttl === null) {
            return $this->set($item->getKey(), serialize($item->get()));
        }
        
        return $this->setex($item->getKey(), $ttl, serialize($item->get()));
```